ethereum-explorer/src/map_contract_events.rs

```diff
@@ -0,0 +1,73 @@
+use crate::pb::eth::event::v1::Events;
+use crate::pb::eth::event::v1::Event;
+use substreams_ethereum::pb::eth::v2::Block;
+use substreams::Hex;
+use substreams_ethereum::pb::eth::v2::Log;
+use substreams_ethereum::pb::eth::v2::TransactionTrace;
+use substreams::errors::Error;
+use crate::util;
+
+#[substreams::handlers::map]
+fn map_contract_events(contract_address: String, blk: Block) -> Result<Events, Error> {
+   let error = verify_parameter(&contract_address);
+   if error.is_some() {
+    return Err(error.unwrap());
+   }
```

Comment:
Rust uses the `Result` enum to deal with errors, when using it, you gain access to early exit syntax `?`. This code can be re-written:

```suggestion
   verify_parameter(&contract_address)?;
```

If `verify_parameter` is modified (see edit suggestion there for the changes to it)

Reply:
If could actually be:

```rust
let address = decode_address(&contract_address)?;
```

And even better (but not working out of the box today), no need to introduce `decode_address` that just wraps `Hex::decode`:

```rust
let address = Hex::decode(&contract_address).with_context(|| "invalid address param {}", &contract_address)?;
```